components/raftstore/src/store/peer.rs

```diff
@@ -2429,6 +2600,26 @@ where
 
         poll_ctx.raft_metrics.propose.normal += 1;
 
+        if self.get_store().applied_index_term() == self.term() {
+            // Only when applied index's term is equal to current leader's term, the information
+            // in epoch checker is up to date and can be used to check epoch.
```

Comment:
Is this check needed? The information in epoch checker will be reset when term change, so the information is either empty or all come from current term.

Reply:
It's needed because the information lacks the proposals from the previous leader which may lead to wrong checking.

Reply:
Missing proposals from the previous leader may lead to false-positive (i.e request will not pass the check at apply worker),  but IMO, rejecting admin request that from current term won't help, in contrast, we should accept them to reduce false-positive.

Reply:
If we accept them, we should update the `cmd_epoch_checker` but it's impossible to check whether this cmd can be applied which breaks correctness dependency.

Reply:
Any uncommitted cmd may not be applied due to leadership change, even cmd proposed after the leader has applied on its term.